packages/network/src/createTxQueue.ts

```diff
@@ -227,14 +230,31 @@ export function createTxQueue<C extends Contracts>(
           error &&
           (("code" in error && error.code === "NONCE_EXPIRED") ||
             JSON.stringify(error).includes("transaction already imported"));
+        const incorrectGasEstimation =
+          error &&
+          (JSON.stringify(error).includes("gas required exceeds allowance") ||
+            JSON.stringify(error).includes("intrinsic gas too low"));
         console.log(
           `[TXQueue] TX Sent (error=${!!error}, isMutationError=${!!isNonViewTransaction} incNonce=${!!shouldIncreaseNonce} resetNonce=${!!shouldResetNonce})`
         );
-        // Nonce handeling
-        if (shouldIncreaseNonce) incNonce();
+
```

Comment:
i think if the transaction hit the mempool and wasn't a view function call, we need to increase the nonce in any case. We should verify that though.

Reply:
ah yeah this is right